torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -1011,13 +1012,18 @@ class TORCH_CUDA_CU_API TensorIndex final : public Val {
     return indices_;
   }
 
+  const std::vector<bool>& rootDomainIndexMap() const {
+    return root_domain_index_map_;
+  }
+
   const TensorView* view() const {
     return view_;
   }
 
  private:
   const TensorView* view_ = nullptr;
   std::vector<Val*> indices_;
+  std::vector<bool> root_domain_index_map_;
```

Comment:
What does this bool vector mean? Can you please add a comment?

Reply:
I need to find the full size of the inner-most, merged root domain.
A root domain has a zero index when it is merged with another domain.
I use this bool vector to mark which root domain indices are zero.

Is there a more robust way to find which root domains are merged together?